include/deal.II/lac/trilinos_tpetra_vector.templates.h

```diff
@@ -44,6 +44,37 @@ namespace LinearAlgebra
 {
   namespace TpetraWrappers
   {
+    namespace internal
+    {
+      template <typename Number, typename MemorySpace>
+      VectorReference<Number, MemorySpace>::operator Number() const
+      {
+        // Get the local index
+        const TrilinosWrappers::types::int_type local_index =
+          vector.vector->getMap()->getLocalElement(
+            static_cast<TrilinosWrappers::types::int_type>(index));
+
+#  if DEAL_II_TRILINOS_VERSION_GTE(14, 0, 0)
+        Assert((local_index != Teuchos::OrdinalTraits<int>::invalid()),
```

Comment:
If `local_index` is of type `TrilinosWrappers::types::int_type`, do you not need to also compare against `Teuchos::OrdinalTraits<TrilinosWrappers::types::int_type>::invalid()`?

Reply:
Or perhaps by `Teuchos::OrdinalTraits<LocalOrdinal>::invalid()` where `LocalOrdinal` is the type used by the `Map` class here -- because `LocalOrdinal` is what the function returns: https://docs.trilinos.org/dev/packages/tpetra/doc/html/classTpetra_1_1Map.html#a8f669db2e7652ed6e50ee1f15711a8f7